tabpfn/utils.py

```diff
@@ -172,19 +171,30 @@ def nan_handling_missing_for_no_reason_value(set_value_to_nan=0.0):
 def nan_handling_missing_for_a_reason_value(set_value_to_nan=0.0):
     return get_nan_value(float('inf'), set_value_to_nan)
 
+def torch_masked_mean_and_std(x, mask, axis=0):
+    """
+    Returns the mean and std of a torch tensor and only considers the elements, where the mask is true
+    """
+    num = torch.where(mask, torch.full_like(x, 1), torch.full_like(x, 0)).sum(dim=axis)
+    value = torch.where(mask, x, torch.full_like(x, 0)).sum(dim=axis)
+    mean = value / num
+    mean_broadcast = torch.repeat_interleave(mean.unsqueeze(axis), x.shape[axis], dim=axis)
+    quadratic_difference_from_mean = torch.square(torch.where(mask, mean_broadcast - x, torch.full_like(x, 0)))
+    return mean, torch.sqrt(torch.sum(quadratic_difference_from_mean, dim=axis) / (num - 1))
+
```

Comment:
Could this be split into two functions and then be re-used by torch_nanmean/nanstd?